src/server/node_manager.c

```diff
@@ -7111,6 +7111,25 @@ set_nodes(void *pobj, int objtype, char *execvnod_in, char **execvnod_out, char
 				return (PBSE_UNKNODE);
 			}
 
+                        if ((pnode->nd_state & (INUSE_DOWN | INUSE_STALE)) && (svr_init == FALSE)) {
+
+				if(objtype == JOB_OBJECT) {
+                                   free(phowl);
+                                   free(execvncopy);
+                                   return (PBSE_BAD_NODE_STATE);
+				} 
+				else {
+					if(presv->ri_qs.ri_state == RESV_UNCONFIRMED) {
+						free(phowl);
+                	                	free(execvncopy);
+                        	      		return (PBSE_BAD_NODE_STATE);
```

Comment:
This is fine for now, but maintenance reservations are in the works.  Those will want to be confirmed on down nodes.  I'd just degrade all reservations.  Either that or we go with what you have now, and when maintenance reservations come in, they change this to allow it to work.

Reply:
Thanks for your suggestions @bhroam 

I am more inclined towards the current approach rather degrading the reservation request.  As you have stated perfectly while implementing the maintenance reservation on down nodes they should consider this race case condition too.  Here we are making the behavior intact to the normal case that if the scheduler couldn't allocate the requested nodes, reservation request will be rejected.

In the case of standing reservations occurrence, we are retaining the nodes in set_nodes() under DEGRADED state, so that whenever a node comes up, the server gets notified and trigger the vnode_available() for making state transition from DEGRADED to CONFIRMATION.

Reply:
@bremanandjk - wouldn't the logic of vnode_available() be applicable for advance reservations as well?

Reply:
This race condition and maintenance reservations are two different use cases.  With this race condition, it is possible for the scheduler to accidentally assign down/stale nodes to a reservation.  In the case of maintenance reservations, they want to assign down nodes to a reservation.  I just figured we could do something that works for both cases here.  If not, you are writing code that will force the person that implements maintenance reservations to change it.

Reply:
You are right @bhroam. I agree, its a good opportunity to make this solution works for both cases(race condition and maintenance reservations). 

Right now, I will mark this PR in WIP and push another commit.

I would greatly appreciate it if you kindly give me some suggestions to move on this.

Reply:
I think we should just mark all reservations that fall in this case degraded and attempt to reconfirm them shortly.